target/cheri-common/cheri_utils.h

```diff
@@ -202,30 +233,41 @@ static inline bool cap_is_unsealed(const cap_register_t *c)
 {
     // TODO: how should we treat the other reserved types? as sealed?
     // TODO: what about untagged capabilities with out-of-range otypes?
+#ifdef TARGET_AARCH64
+    return cap_get_otype_unsigned(c) == CAP_OTYPE_UNSEALED;
+#else
     _Static_assert(CAP_MAX_REPRESENTABLE_OTYPE == CAP_OTYPE_UNSEALED, "");
     if (c->cr_tag) {
         cheri_debug_assert(cap_get_otype_unsigned(c) <=
                            CAP_MAX_REPRESENTABLE_OTYPE);
     }
     return cap_get_otype_unsigned(c) >= CAP_OTYPE_UNSEALED;
+#endif
 }
 
 static inline void cap_set_sealed(cap_register_t *c, uint32_t type)
 {
+#ifdef TARGET_AARCH64
+    type &= CAP_CC(FIELD_OTYPE_MASK_NOT_SHIFTED);
+#endif
     assert(c->cr_tag);
     assert(cap_get_otype_unsigned(c) == CAP_OTYPE_UNSEALED &&
            "should not use this on caps with reserved otypes");
     assert(type <= CAP_LAST_NONRESERVED_OTYPE);
+#ifndef TARGET_AARCH64
     _Static_assert(CAP_LAST_NONRESERVED_OTYPE < CAP_OTYPE_UNSEALED, "");
+#endif
     CAP_cc(update_otype)(c, type);
 }
 
 static inline void cap_set_unsealed(cap_register_t *c)
 {
     assert(c->cr_tag);
+#ifndef TARGET_AARCH64
     assert(cap_is_sealed_with_type(c));
     assert(cap_get_otype_unsigned(c) <= CAP_LAST_NONRESERVED_OTYPE &&
```

Comment:
This should still hold if you use cap_is_sealed_with_type, or inline the subset needed here

Reply:
Ah yeah, I think I updated the call that asserts here to use a direct otype update.